<a href="https://colab.research.google.com/github/TanChen168/ExploratoryVizWeek3/blob/main/AverageHeight.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [10]:
import pandas as pd

filename = '/content/sample_data/athleteEventsNoPersonal.csv'

df = pd.read_csv(filename)

df

ID   Age  ...                                     Event Medal
0           5  21.0  ...          Speed Skating Women's 500 metres   NaN
1           5  21.0  ...        Speed Skating Women's 1,000 metres   NaN
2           5  25.0  ...          Speed Skating Women's 500 metres   NaN
3           5  25.0  ...        Speed Skating Women's 1,000 metres   NaN
4           5  27.0  ...          Speed Skating Women's 500 metres   NaN
...       ...   ...  ...                                       ...   ...
40611  135567  28.0  ...    Rowing Men's Lightweight Coxless Fours   NaN
40612  135570  27.0  ...  Ski Jumping Men's Large Hill, Individual   NaN
40613  135570  27.0  ...        Ski Jumping Men's Large Hill, Team   NaN
40614  135571  30.0  ...                      Bobsleigh Men's Four   NaN
40615  135571  34.0  ...                      Bobsleigh Men's Four   NaN

[40616 rows x 12 columns]

1.a How many rows? - 40616


1.b How many columns? - 12


1.c Which columns have missing values? - Age, Height, Medal got missing values. Athelets may not win medals in all events that's why it has missing values.

In [2]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40616 entries, 0 to 40615
Data columns (total 12 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   ID      40616 non-null  int64  
 1   Age     39143 non-null  float64
 2   Height  31615 non-null  float64
 3   Team    40616 non-null  object 
 4   NOC     40616 non-null  object 
 5   Games   40616 non-null  object 
 6   Year    40616 non-null  int64  
 7   Season  40616 non-null  object 
 8   City    40616 non-null  object 
 9   Sport   40616 non-null  object 
 10  Event   40616 non-null  object 
 11  Medal   5917 non-null   object 
dtypes: float64(2), int64(2), object(8)
memory usage: 3.7+ MB


1.d. How many entries correspond to the city of London? - 3370 entries

In [3]:
city = df.loc[df['City'] == 'London', :]
city

ID   Age  ...                                         Event   Medal
36         54   NaN  ...                      Boxing Men's Heavyweight     NaN
42         67  26.0  ...               Athletics Women's Javelin Throw     NaN
66        155  34.0  ...      Wrestling Men's Heavyweight, Greco-Roman     NaN
69        159  19.0  ...            Synchronized Swimming Women's Team     NaN
72        185  31.0  ...                Fencing Men's Foil, Individual     NaN
...       ...   ...  ...                                           ...     ...
40577  135489  22.0  ...        Swimming Women's 200 metres Backstroke  Silver
40578  135489  22.0  ...  Swimming Women's 4 x 100 metres Medley Relay     NaN
40590  135517  22.0  ...          Swimming Men's 100 metres Backstroke     NaN
40595  135528  28.0  ...                       Badminton Men's Singles     NaN
40609  135558  32.0  ...                   Volleyball Men's Volleyball     NaN

[3370 rows x 12 columns]

1.e. What age is the youngest athlete in our sample data? - 11

In [5]:
age = df['Age'].min()
age

11.0

2.a. How many athletes who participated in the Sport "Tug-Of-War" were from Team "Sweden" in our sample dataset? - 6

In [19]:
City_filter = df['NOC'] == 'SWE'
Event_filter = df['Sport']=='Tug-Of-War'
Sweden_ToF = df.loc[City_filter & Event_filter, :]
Sweden_ToF

ID   Age  Height  ...       Sport                        Event   Medal
10023  34205  22.0     NaN  ...  Tug-Of-War  Tug-Of-War Men's Tug-Of-War     NaN
12681  43105  23.0     NaN  ...  Tug-Of-War  Tug-Of-War Men's Tug-Of-War  Bronze
13015  44087  29.0   182.0  ...  Tug-Of-War  Tug-Of-War Men's Tug-Of-War  Bronze
16250  55169  23.0     NaN  ...  Tug-Of-War  Tug-Of-War Men's Tug-Of-War     NaN
19048  64266  21.0     NaN  ...  Tug-Of-War  Tug-Of-War Men's Tug-Of-War     NaN
20550  68647  26.0   190.0  ...  Tug-Of-War  Tug-Of-War Men's Tug-Of-War  Bronze

[6 rows x 12 columns]

2.b. How many Gold medals were awarded to athletes who were over 40 years old? - 66 gold medals

In [23]:
gold_filter = df['Medal'] == 'Gold'
yo40_filter = df['Age'] > 40
Gold40yo = df.loc[gold_filter & yo40_filter, :]
Gold40yo

ID   Age  ...                                        Event Medal
740      2735  41.0  ...  Shooting Men's Rapid-Fire Pistol, 25 metres  Gold
1103     4029  46.0  ...                              Polo Men's Polo  Gold
1421     4993  41.0  ...                      Sailing Mixed 10 metres  Gold
3040    10769  49.0  ...          Sailing Mixed Three Person Keelboat  Gold
3270    11617  44.0  ...                       Sailing Mixed 7 metres  Gold
...       ...   ...  ...                                          ...   ...
38916  130271  43.0  ...                      Sailing Mixed 12 metres  Gold
38952  130381  42.0  ...                   Archery Women's Individual  Gold
39193  130936  56.0  ...   Shooting Men's Running Target, Double Shot  Gold
39197  130936  60.0  ...           Art Competitions Mixed Sculpturing  Gold
39365  131533  58.0  ...                   Archery Women's Team Round  Gold

[66 rows x 12 columns]

3. Determine the average height of:
     a. All rows included - 175.38

In [25]:
df['Height'].mean()

175.3811165585956

3. Determine the average height of:
     b. All rows included, but grouped by Event

In [27]:
df.groupby(['Event']).mean()

ID  ...         Year
Event                                                          ...             
Alpine Skiing Men's Combined                     64389.042254  ...  1986.676056
Alpine Skiing Men's Downhill                     68818.789474  ...  1978.200000
Alpine Skiing Men's Giant Slalom                 66300.021277  ...  1983.744681
Alpine Skiing Men's Slalom                       65713.417722  ...  1983.071730
Alpine Skiing Men's Super G                      63096.333333  ...  1999.078431
...                                                       ...  ...          ...
Wrestling Women's Flyweight, Freestyle           49759.500000  ...  2011.600000
Wrestling Women's Heavyweight, Freestyle         84851.222222  ...  2012.000000
Wrestling Women's Light-Heavyweight, Freestyle  108636.500000  ...  2016.000000
Wrestling Women's Lightweight, Freestyle         66110.333333  ...  2011.666667
Wrestling Women's Middleweight, Freestyle        87807.600000  ...  2012.266667

[726 rows x 4 columns]

3.c. Bonus

In [31]:
one_per_athlete = df.drop_duplicates(subset = 'ID', keep = 'first')
one_per_athlete.groupby(['Event']).mean()

ID  ...         Year
Event                                                          ...             
Alpine Skiing Men's Combined                     52150.083333  ...  1956.500000
Alpine Skiing Men's Downhill                     68067.735043  ...  1975.487179
Alpine Skiing Men's Giant Slalom                 67346.921053  ...  1989.921053
Alpine Skiing Men's Slalom                       69501.322581  ...  1988.258065
Alpine Skiing Men's Super G                      61497.692308  ...  1995.000000
...                                                       ...  ...          ...
Wrestling Women's Flyweight, Freestyle           50271.000000  ...  2010.500000
Wrestling Women's Heavyweight, Freestyle         79878.285714  ...  2011.428571
Wrestling Women's Light-Heavyweight, Freestyle  108636.500000  ...  2016.000000
Wrestling Women's Lightweight, Freestyle         64653.300000  ...  2011.600000
Wrestling Women's Middleweight, Freestyle        86701.500000  ...  2010.800000

[609 rows x 4 columns]